qiskit_experiments/data_processing/base.py

```diff
@@ -0,0 +1,99 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Defines the steps that can be used to analyse data."""
+
+from abc import ABCMeta, abstractmethod
+from typing import Any, Dict, List
+
+from qiskit_experiments.data_processing.exceptions import DataProcessorError
+
+
+class DataAction(metaclass=ABCMeta):
+    """
+    Abstract action which is a single action done on measured data to process it.
+    Each subclass of DataAction must define the type of data that it accepts as input
+    using decorators.
+    """
+
+    def __init__(self):
+        """Create new data analysis routine."""
+        self._child = None
+        self._accepted_inputs = []
```

Comment:
Based on subclass implementation it seems to be class variable.
```python
class DataAction(ABCMeta):
   __input_data_type__ = list()
   __output_data_type__ = None
```

Reply:
What if the user wants to add his own acpeted data format? E.g. I want to use to `ToReal` but before `ToReal` I apply a custom node that outputs something else than memory, say `my_new_data_key`. With the current implementation the user can do
```python
to_real = ToReal(1e-3)
to_real.add_accepted_input('my_new_data_key')
processor.append(to_real)
```
So I think we can keep `_accepted_inputs` as is but `node_output` can be a class variable.

Reply:
Done in 681546f